apps/server/src/modules/user/service/user-query.type.ts

```diff
@@ -0,0 +1,6 @@
+import { UserDO } from '@shared/domain/domainobject/user.do';
+
+export type UserQuery = Partial<Pick<UserDO, 'schoolId'>> & {
+	isOutdated?: boolean;
+	hasPreviousExternalId?: boolean;
```

Comment:
lastLoginSystemChange should be the marker we check for to find a migrated user. PreviousExternalId should only be used for rollback scenarios (or to avoid them, like in the ldap sync). This would also align with potential future migrations from non-external systems (like local) where we would not have a previousExternalId. Also this field would be already set if a user makes a second migration later on, which would nullify the option to distinguish migrated from non-migrated users. This was the conclusion from a discussion with our teammate, who researches the extension to non-oauth-migrations